client/containers/Pub/PubDocument/Review/ReviewModal.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from 'react';
+import { Button, NonIdealState } from '@blueprintjs/core';
+
+import { apiFetch } from 'client/utils/apiFetch';
+import { DialogLauncher } from 'components';
+import { DocJson } from 'types';
+import { usePageContext } from 'utils/hooks';
+
+import ReviewerDialog from './ReviewerDialog';
+
+type Props = {
+	isLoading: boolean;
+	pubData: any;
+	communityData: any;
+	updatePubData: any;
+	reviewDoc: DocJson;
+	setIsLoading: any;
+};
+
+const ReviewModal = (props: Props) => {
+	const { isLoading, pubData, communityData, updatePubData, reviewDoc, setIsLoading } = props;
+	const {
+		scopeData: { activePermissions },
+		loginData: { fullName },
+	} = usePageContext();
+	const [reviewTitle, setReviewTitle] = useState('Untilted Review');
+	const [reviewerName, setReviewerName] = useState(fullName || 'anonymous');
+	const [createError, setCreateError] = useState(undefined);
+	const isUser = !!(activePermissions.canEdit || fullName);
+	const redirectUrl = (review) =>
+		isUser ? `/dash/pub/${pubData.slug}/reviews/${review.number}` : `/signup`;
+	const saveReviewerName = (review) => {
+		return apiFetch
+			.post('/api/reviewer', {
+				reviewId: review.id,
+				name: reviewerName,
+				permissions: activePermissions,
+			})
+			.catch((err) => {
+				setIsLoading(false);
+				setCreateError(err);
+			});
+	};
+	const createReviewDoc = () => {
+		setIsLoading(true);
+		apiFetch
+			.post('/api/reviews', {
+				communityId: communityData.id,
+				pubId: pubData.id,
+				review: reviewDoc,
+				title: reviewTitle,
+				permissions: activePermissions,
+			})
+			.then((review) => {
+				saveReviewerName(review);
+				updatePubData((currentPubData) => {
+					return {
+						reviews: [...currentPubData.reviews, review],
+					};
+				});
+				setIsLoading(false);
+				window.location.href = redirectUrl(review);
```

Comment:
Do we ever let anonymous users know that they'll be redirected to the signup screen when their review is submitted? It may be a little bewildering to people why they ended up there without any notice.